test/lib/sstable_test_env.hh

```diff
@@ -110,8 +110,7 @@ public:
         return reusable_sst(std::move(schema), dir, generation).then([] (auto ptr) { return make_ready_future<>(); });
     }
 
-    template <typename Func>
-    static inline auto do_with(Func&& func, test_env_config cfg = {}) {
+    static inline auto do_with(noncopyable_function<future<> (test_env&)> func, test_env_config cfg = {}) {
```

Comment:
We don't need the `auto` return type anymore either, can just specify it as `future<>`.

Reply:
Actually we do, since actual argument funcs also return `future<T>` (and so this line is broken)